pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -117,20 +117,31 @@ protected synchronized void deleteSegmentFromPropertyStoreAndLocal(String tableN
     }
 
     int nSegmentsDeleted = 0;
-    if (segmentsToDelete.size() > 0) {
+    if (!segmentsToDelete.isEmpty()) {
+      List<String> propStorePathList = new ArrayList<>(segmentsToDelete.size());
       for (String segmentId : segmentsToDelete) {
         String segmentPropertyStorePath = ZKMetadataProvider.constructPropertyStorePathForSegment(tableName, segmentId);
-        LOGGER.info("Trying to delete segment : {} from Property store.", segmentId);
-        boolean deletionFromPropertyStoreSuccessful = true;
-        if (_propertyStore.exists(segmentPropertyStorePath, AccessOption.PERSISTENT)) {
-          deletionFromPropertyStoreSuccessful = _propertyStore.remove(segmentPropertyStorePath, AccessOption.PERSISTENT);
+        propStorePathList.add(segmentPropertyStorePath);
+      }
+
+      boolean[] status = _propertyStore.remove(propStorePathList, AccessOption.PERSISTENT);
```

Comment:
I would rename this to `deleteSuccessful` so that the condition becomes `if (!deleteSuccessful[i])` later on.